zipline/gens/tradesimulation.py

```diff
@@ -145,6 +148,16 @@ def every_bar(dt_to_use, current_data=self.current_data,
 
         def once_a_day(midnight_dt, current_data=self.current_data,
                        data_portal=self.data_portal):
+
+            perf_tracker = algo.perf_tracker
```

Comment:
Since `every_bar` and `once_a_day` are all closures, this could be moved even further up to where `algo = self.algo` is defined. Though, not blocking for this PR.